lib/wallets/wallet/impl/solana_wallet.dart

```diff
@@ -0,0 +1,362 @@
+
+  NodeModel? _solNode;
+
+  Future<Ed25519HDKeyPair> _getKeyPair() async {
+    return Ed25519HDKeyPair.fromMnemonic(await getMnemonic(), account: 0, change: 0);
+  }
+
+  Future<Address> _getCurrentAddress() async {
+    var addressStruct = Address(
+        walletId: walletId,
+        value: (await _getKeyPair()).address,
+        publicKey: List<int>.empty(),
+        derivationIndex: 0,
+        derivationPath: null,
+        type: cryptoCurrency.coin.primaryAddressType,
+        subType: AddressSubType.unknown);
+    return addressStruct;
+  }
+
+  Future<int> _getCurrentBalanceInLamports() async {
+    var rpcClient = RpcClient("${getCurrentNode().host}:${getCurrentNode().port}");
+    var balance = await rpcClient.getBalance((await _getKeyPair()).address);
+    return balance.value;
+  }
+
+  @override
+  FilterOperation? get changeAddressFilterOperation =>
+      throw UnimplementedError();
+
+  @override
+  Future<void> checkSaveInitialReceivingAddress() async {
+    try {
+      var address = (await _getKeyPair()).address;
+
+      await mainDB.updateOrPutAddresses([
+        Address(
+            walletId: walletId,
+            value: address,
+            publicKey: List<int>.empty(),
+            derivationIndex: 0,
+            derivationPath: null,
+            type: cryptoCurrency.coin.primaryAddressType,
+            subType: AddressSubType.unknown)
+      ]);
+    } catch (e, s) {
+      Logging.instance.log(
+        "$runtimeType  checkSaveInitialReceivingAddress() failed: $e\n$s",
+        level: LogLevel.Error,
+      );
+    }
+  }
+
+  @override
+  Future<TxData> prepareSend({required TxData txData}) async {
+    try {
+      if (txData.recipients == null || txData.recipients!.length != 1) {
+        throw Exception("$runtimeType prepareSend requires 1 recipient");
+      }
+
+      Amount sendAmount = txData.amount!;
+
+      if (sendAmount > info.cachedBalance.spendable) {
+        throw Exception("Insufficient available balance");
+      }
+
+      int feeAmount;
+      var currentFees = await fees;
+      switch (txData.feeRateType) {
+        case FeeRateType.fast:
+          feeAmount = currentFees.fast;
+          break;
+        case FeeRateType.slow:
+          feeAmount = currentFees.slow;
+          break;
+        case FeeRateType.average:
+        default:
+          feeAmount = currentFees.medium;
+          break;
+      }
+
+      // Rent exemption of Solana
+      final rpcClient = RpcClient("${getCurrentNode().host}:${getCurrentNode().port}");
+      final accInfo = await rpcClient.getAccountInfo((await _getKeyPair()).address);
+      final minimumRent = await rpcClient.getMinimumBalanceForRentExemption(accInfo.value!.data.toString().length);
+      if (minimumRent > ((await _getCurrentBalanceInLamports()) - txData.amount!.raw.toInt() - feeAmount)) {
+        throw Exception("Insufficient remaining balance for rent exemption, minimum rent: ${minimumRent / pow(10, cryptoCurrency.fractionDigits)}");
```

Comment:
What if we have less than the minimum rent amount?  Can we still send?

Reply:
No, the library and the network doesn't allow.

Reply:
I thought you should be able to reclaim your rent eventually, no?

I'm not very educated on Solana tho.